packages/arco-lib/src/components/Table.tsx

```diff
@@ -213,100 +270,113 @@ export const Table = implementRuntimeComponent({
     });
   }, [selectedRowKeys, mergeState]);
 
-  const inputRef = useRef(null);
-
-  const columns = cProps.columns!.map((column, i) => {
-    const newColumn: ColumnProperty = { ...column };
-    if (newColumn.filter) {
-      newColumn.filterDropdown = ({
-        filterKeys,
-        setFilterKeys,
-        confirm,
-      }: filterDropdownParam) => {
-        return (
-          <div className="arco-table-custom-filter">
-            <Input.Search
-              ref={inputRef}
-              searchButton
-              placeholder="Please input and enter"
-              value={filterKeys?.[0] || ''}
-              onChange={value => {
-                setFilterKeys && setFilterKeys(value ? [value] : []);
-              }}
-              onSearch={() => {
-                confirm && confirm();
-              }}
-            />
-          </div>
-        );
-      };
-    }
-
-    newColumn.render = (ceilValue: any, record: any, index: number) => {
-      const evaledColumn: ColumnProperty = services.stateManager.deepEval(column, true, {
-        [LIST_ITEM_EXP]: record,
-      });
-      const value = record[evaledColumn.dataIndex];
-
-      let colItem;
-
-      switch (evaledColumn.type) {
-        case 'button':
-          const handleClick = () => {
-            const rawColumn = (component.properties.columns as ColumnProperty[])[i];
-            if (!rawColumn.btnCfg) return;
-            const evaledButtonConfig = services.stateManager.deepEval(
-              rawColumn.btnCfg,
-              true,
-              { [LIST_ITEM_EXP]: record }
+  useEffect(() => {
+    setColumns(
```

Comment:
Why make columns to state and put it in useEffect?

Reply:
Once the width has been changed by dragging, the width of the column needs to be changed by callback

Reply:
oh,I was wrong, it's just so that when you change the width in inspect it takes effect immediately